rq/plan.sh

```diff
@@ -1,24 +1,23 @@
 pkg_name=rq
 pkg_origin=core
-pkg_version=0.9.2
-pkg_source=https://github.com/dflemstr/rq/releases/download/v${pkg_version}/${pkg_name}-x86_64-unknown-linux-musl
-pkg_shasum=01718bd4e52e5139d050dabbea308566a20c54d62a46fd28ab900819acc31c8b
+pkg_version=0.10.4
+pkg_filename=record-query-v${pkg_version}-x86_64-unknown-linux-gnu.tar.gz
+pkg_source=https://github.com/dflemstr/rq/releases/download/v${pkg_version}/${pkg_filename}
+pkg_shasum=1b9abedd7acca1269ac1c4765d93bab73777feca9a04358ea4f9c9659bf51e62
 pkg_maintainer="The Habitat Maintainers <humans@habitat.sh>"
 pkg_description="Record Query - A tool for doing record analysis and transformation"
 pkg_upstream_url=https://github.com/dflemstr/rq
 pkg_license=('Apache-2.0')
 pkg_bin_dirs=(bin)
+pkg_deps=(core/glibc core/gcc-libs)
+pkg_build_deps=(core/patchelf)
 
 do_unpack() {
-  return 0
-}
-
-do_prepare() {
-  return 0
-}
-
-do_verify() {
-  return 0
+  do_default_unpack
+  # the source tarball contains a cross-compiled binary in a parent directory
+  # with a vague name. here we'll rename it to the hab standard pkg_dirname
+  # instead of setting pkg_dirname to this vague name.
+  mv x86_64-unknown-linux-gnu "$pkg_dirname"
```

Comment:
I'm not sure if this is a `hab` version difference, but to build this locally I needed to change this to:

```
mv $HAB_CACHE_SRC_PATH/x86_64-unknown-linux-gnu "$HAB_CACHE_SRC_PATH/$pkg_dirname"
```

Reply:
This seems to be some new behavior in 0.50.3. The working directory of `do_unpack()` has historically been `$HAB_CACHE_SRC_PATH` (`/hab/cache/src`).

I added the following to the beginning of this plan's `do_unpack`:

```
build_line "This is where we're running unpack: $(pwd)"
```

For the first build in a clean studio, I saw the error and I note that the working directory is `/src/rq`.

```
   rq: This is where we're running unpack: /src/rq
   rq: Unpacking record-query-v0.10.4-x86_64-unknown-linux-gnu.tar.gz
mv: cannot stat 'x86_64-unknown-linux-gnu': No such file or directory
```

A second build run immediately after has a different working directory and succeeds:

```
   rq: This is where we're running unpack: /hab/cache/src
   rq: Unpacking record-query-v0.10.4-x86_64-unknown-linux-gnu.tar.gz
   rq: Setting build environment
```

So, I _can_ update this plan to be explicit about the full path to the directory to rename. I wonder, though, if that would masking over some new unexpected behavior in the expected working directories of the plan build functions.

Reply:
@robbkidd Yeah, I was almost surely on an older version.  I think for core-plans it is OK for us to assume "latest hab" for *build time* and trust the habitat maintainers to keep the appropriate about of *run time* backwards compat.

Reply:
@stevendanna Do you have a recommendation in light of this? I suppose I could wrap the `mv` command in a `pushd $HAB_CACHE_SRC_PATH ; mv ... ; popd`. But I'm left wondering what changed that the first build has a different working directory for `do_unpack` (and maybe other callbacks) than second and subsequent builds.

Reply:
@robbkidd I am also wondering what changed. I added it to my "hab things to look into" todo list but probably won't get to it until later tomorrow.

For now, I would probably lean towards the thing that will work regardless of the working directly, which is directly using $HAB_CACHE_PATH to reference the source and target directories by absolute paths.